src/components/navbar/Navbar/ProfileModal.tsx

```diff
@@ -243,12 +258,14 @@ function PrivateKeyContent() {
 }
 
 function LogoutContent({ setCurrentState }: ContentProps) {
-  const logout = useMyAccount((state) => state.logout)
+  const { logout, getPrivateKey } = useMyAccount((state) => state)
   const sendEvent = useSendEvent()
 
-  const onShowPrivateKeyClick = () => {
-    sendEvent('click no_show_me_my_private_key_button')
-    setCurrentState('private-key')
+  const onShowPrivateKeyClick = async () => {
+    await getPrivateKey().then(() => {
```

Comment:
and this kind of thing will be solved if you just get the private key when you login.

I think it won't be any security issue, as it will be mostly same as your current implementation, just get the private key only once in the start